lang/cs/Org.Apache.REEF.Client/Local/TestRunner/FileWritingAssert/TestResult.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using Newtonsoft.Json;
+using Org.Apache.REEF.Client.API.Testing;
+
+namespace Org.Apache.REEF.Client.Local.TestRunner.FileWritingAssert
+{
+    internal sealed class TestResult : ITestResult, IEquatable<TestResult>
+    {
+        private readonly ISet<AssertResult> _results;
+
+        public TestResult(ISet<AssertResult> results)
+        {
+            _results = results;
+        }
+
+        public TestResult() : this(new HashSet<AssertResult>())
+        {
+            // Intentionally empty
+        }
+
+        public int NumberOfFailedAsserts
+        {
+            get
+            {
+                return _results.Count(_ => _.IsFalse);
```

Comment:
Here and everywhere: I think we can do simply `_results.Count(IsFalse)` etc.

Reply:
Hmm, I can't get that to work. I believe this works if `IsFalse` is a method in the current context, not of the objects in the collection?